src/browser.ts

```diff
@@ -123,26 +199,22 @@ async function flushBuffered(analytics: Analytics): Promise<void> {
  * Also Ensures events can be registered before library initialization.
  * This is important so users can register to 'initialize' and any events that may fire early during setup.
  */
-function flushPreBuffer(analytics: Analytics): void {
-  const wa = window.analytics
-  const buffered =
-    // @ts-expect-error
-    wa && wa[0] ? [...wa] : []
 
-  const anon = buffered.find(([op]) => op === 'setAnonymousId')
-  if (anon) {
-    const [, id] = anon
-    analytics.setAnonymousId(id)
+async function flushPreBuffer(
```

Comment:
I think some opportunity for optimization here (I know it is old code), but given `on` and `setAnonymousId` aren't async we can filter for both `setAnonymousId` and `on` in one go, and then call all of them sequentially in a `forEach`

Reply:
I made it a bit more functional instead of using a foreach -- it's still O(n) -- let me know your thoughts.